queries/vhdl/highlights.scm

```diff
@@ -0,0 +1,2397 @@
+((return_statement
+   "return" @_c @error @error.scope.return) @error.region
+   (#not-has-ancestor? @_c function_body procedure_body))
+
+; Statements after return statement are unreachable (dead code)
+((return_statement) . (_)* @error.region.unreachble)
+; }}}
+; 11.2 Block statement {{{
+((block_statement
+         . (label [(identifier)  (extended_identifier)]  @_h @error @error.repeated.name)
+   at_end:        [(simple_name) (extended_simple_name)] @_t @error @error.repeated.name .)
+           (#not-eq? @_h @_t))
+; }}}
+; 11.3 Process statement {{{
+((process_statement
+         . (label [(identifier)  (extended_identifier)]  @_h @error @error.repeated.name)
+   at_end:        [(simple_name) (extended_simple_name)] @_t @error @error.repeated.name .)
+           (#not-eq? @_h @_t))
+; }}}
+; 11.6 Concurrent signal assignments {{{
+(concurrent_signal_assignment
+   target: (aggregate
+               (named_element_association "=>") @error) @error.region)
+; }}}
+; 11.7 Component instantiation statements {{{
+(component_instantiation_statement
+   (map_aspect [
+      (generic_clause     (semicolon) @error @error.unexpected.semicolon)
+      (generic_map_aspect (semicolon) @error @error.unexpected.semicolon)
+      (port_clause        (semicolon) @error @error.unexpected.semicolon)
+      (port_map_aspect    (semicolon) @error @error.unexpected.semicolon)
+
+      (generic_map_aspect [(any) (default)] @error)
+   ])
+)
+; }}}
+; 11.8 Generate statement {{{
+((for_generate_statement
+         . (label [(identifier)  (extended_identifier)]  @_h @error @error.repeated.name)
+   at_end:        [(simple_name) (extended_simple_name)] @_t @error @error.repeated.name .)
+           (#not-eq? @_h @_t))
+
+((if_generate_statement
+         . (label [(identifier) (extended_identifier)] @_h @error @error.repeated.name)
+   at_end: [(simple_name) (extended_simple_name)] @_t @error @error.repeated.name .)
+           (#not-eq? @_h @_t))
+
+((_
+ . (label [(identifier) (extended_identifier)] @_h @error @error.repeated.name )
+   (generate_statement_body
+      at_end: [(simple_name) (extended_simple_name)] @_t @error @error.repeated.name .))
+   (#not-eq? @_h @_t))
+; }}}
+; 12.4 Use clauses {{{
+(selected_name
+   prefix: (selected_name
+             prefix: ((simple_name) @_p
+                      (#vim-match? @_p "^[iI][eE]{3}$"))
+             suffix: ((simple_name) @_s @error @error.illegal.package.ieee
+                      (#not-vim-match? @_s "^\%([nN][uU][mM][eE][rR][iI][cC]_\%([bB][iI][tT]|[sS][tT][dD]\)\%(_\%([uU][nN]\)?[sS][iI][gG][nN][eE][dD]\)?|\%([fF][iI][xX][eE][dD]|[fF][lL][oO][aA][tT]\)\%(_[gG][eE][nN][eE][rR][iI][cC]\)?_[pP][kK][gG]|[fF][iI][xX][eE][dD]_[fF][lL][oO][aA][tT]_[tT][yY][pP][eE][sS]|[sS][tT][dD]_[lL][oO][gG][iI][cC]_\%(1164|[tT][eE][xX][tT][iI][oO]|[aA][rR][iI][tT][hH]|[mM][iI][sS][cC]|\%([uU][nN]\)?[sS][iI][gG][nN][eE][dD]\)|[mM][aA][tT][hH]_\%([rR][eE][aA][lL]|[cC][oO][mM][pP][lL][eE][xX]\)\)$"))
```

Comment:
wouldn't setting the case-insensitve flag `\C` or having  dedicated predicate make in more readable (can't tell whether this will work with all the crazy escaping happening in nvim)

Reply:
I tried using the case insensitive flag but it didn't work. Now, after doing some investigation, I think there is a bug on [neovim/runtime/lua/vim/treesitter/query.lua:175](https://github.com/neovim/neovim/blob/885c2f7281c2f064961e10fc7e55110dff718dc9/runtime/lua/vim/treesitter/query.lua#L175).  The `magic_prefixes[string.sub(str,1,2)]` on line 167 can't be true. 

I used `\%(` instead of `(` because some regex expressions on this query (1) exceeds the hardcoded limit of captures; (2)  don't need the capture -- and `\%(` should be faster.

By the way, if you think that expressions is bad have a look at line 1531.

I will try do some more investigation on the `#match` to see if is actually a bug and then update the regex expressions on the query to increase legibility.

Reply:
Actually, I think creating a new predicate might help writing clearer query and avoid the regex compile overhead.

The code above could look like this:

```
(selected_name
   prefix: (selected_name
             prefix: ((simple_name) @_p
                      (#foobar? @_p "ieee"))
             suffix: ((simple_name) @_s @error @error.illegal.package.ieee
                      (#foobar? @_s "fixed_float_types"
                                    "fixed_pkg"
                                    "float_pkg"
                                    "fixed_generic_pkg"
                                    "float_generic_pkg"
                                    "math_complex"
                                    "math_real"
                                    "numeric_bit"
                                    "numeric_std"
                                    "numeric_bit_unsigned"
                                    "numeric_std_unsigned"
                                    "std_logic_1164"
                                    "std_logic_textio"
                                    "std_logic_misc"
                                    "std_logic_arith"
                                    "std_logic_signed"
                                    "std_logic_unsigned"))) @error.region)
```

Any idea for a name to such predicate? Also, how to tell case sensitive and case insensitive apart?

Reply:
I think it would be worth to try to fix the bug in the regex escaping.
```
((identifier) @foo
(#match? @foo "^\cres$"))
```

Maybe `any-of?` with a boolean parameter for case-sensitive or not?

```
((identifier) @foo
(#any-of? nil "foo" "bar" "baz"))

((identifier) @foo
(#any-of? t "foo" "bar" "baz"))

``